src/main/webapp/app/home/home.component.ts

```diff
@@ -136,23 +136,15 @@ export class HomeComponent implements OnInit, AfterViewChecked {
 
     ngAfterViewChecked() {
         // Only focus the username input once, not on every update
-        if (this.inputFocused || this.loading) {
+        if (this.mainElementFocused || this.loading) {
             return;
         }
 
         // Focus on the main element as soon as it is visible
-        if (!this.passwordLoginDisabled) {
-            const input = this.renderer.selectRootElement('#username', true);
-            if (input) {
-                input.focus();
-                this.inputFocused = true;
-            }
-        } else if (this.saml2Config) {
-            const btn = this.renderer.selectRootElement('#saml2Button', true);
-            if (btn) {
-                btn.focus();
-                this.inputFocused = true;
-            }
+        const mainElement = this.renderer.selectRootElement(this.passwordLoginDisabled ? '#saml2Button' : '#username', true);
+        if (mainElement) {
+            mainElement.focus();
+            this.mainElementFocused = true;
```

Comment:
I think here was a small semantic change. If passwordLoginDisabled == true and saml2Config == false

Reply:
Yes, I am aware of that. The `else if` was not needed as the variable is defined [like so](https://github.com/ls1intum/Artemis/pull/5291/files#diff-c3974a25b9c2e1f16a1c5673271c038fe8f54e66af898a68c131904b4a43f11dR123).